src/OrbitGl/App.cpp

```diff
@@ -349,6 +342,16 @@ void OrbitApp::OnTracepointEvent(orbit_client_protos::TracepointEventInfo tracep
       is_same_pid_as_target);
 }
 
+void OrbitApp::OnModuleUpdate(uint64_t /*timestamp_ns*/, ModuleInfo module_info) {
+  module_manager_->AddOrUpdateModules({module_info});
+  GetMutableCaptureData().mutable_process()->AddOrUpdateModuleInfo(module_info);
+}
+
+void OrbitApp::OnModulesSnapshot(uint64_t /*timestamp_ns*/, std::vector<ModuleInfo> module_infos) {
+  module_manager_->AddOrUpdateModules(module_infos);
+  GetMutableCaptureData().mutable_process()->UpdateModuleInfos(module_infos);
```

Comment:
Besides the process in capture data, there is also `ProcessData* process_` (`GetMutableTargetProcess()`) in OrbitApp (its a pointer to a process living in a `TargetConfiguration` which is owned by orbitmainwindow). 

I think on capture start the target process is copied into capture data. 

I guess the modules of this target process should also be updated (and refresh module list dataview). Maybe on capture end the target process should be overridden with the version from capture data.

Having these 2 processes is not great in the first place. I guess we could also do something about that, but that might entail a lot.  
Let me know what you think, I am also happy to chat about it offline

Reply:
I think the process managed by the app should be updated once capture is done (or pereodically but outside of capture-process thread). It does not have any effect on what we are doing here anyways. I believe we cannot hook/unhook functions during capture right?

Reply:
Since in this case we want to not unload the modules so that after aborting the capture the client is in the consistent state, I will do it in the follow-up change. It ends up being bigger then I expected.